### PR TITLE
JSON API: new options endpoint

### PR DESCRIPTION
**Pull request by roccotripaldi**
An endpoint to update an option.

### CODE REVIEW

**Comment by roccotripaldi** [comment]
Make some changes to this based on a slack conversation with @zinigor and some feedback from @lezama on Phabricator. feel free to punt it off to an other version if it is holding you up.

**Comment by dereksmart** [comment]
@roccotripaldi said to bump to next version

File: json-endpoints/class.wpcom-json-api-update-option-endpoint.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        }
+    }
+
+    function validate_input() {
+        $input = $this->input();
+        if ( ! isset( $input['option_value'] ) || is_array( $input['option_value'] ) ) {
+            return false;
+        }
+        $this->option_value = $input['option_value'];
+        return parent::validate_input();
```
Comment (zinigor) [inline json-endpoints/class.wpcom-json-api-update-option-endpoint.php]: This makes me so happy :)

**Comment by zinigor** [comment]
This PR looks great to me now! The only thing that's left is getting it in on .com side and testing it.

**Comment by dereksmart** [comment]
This is still targeted for 3.8.2, but @lezama is going to work on a jetpack-only version to merge.

**Comment by roccotripaldi** [comment]
Closing in favor of #3139

**Pull request closed without merging**